src/SignalR/clients/csharp/Client.Core/src/HubConnectionBuilder.cs

```diff
@@ -48,6 +52,12 @@ public HubConnection Build()
                 throw new InvalidOperationException($"Cannot create {nameof(HubConnection)} instance. An {nameof(IConnectionFactory)} was not configured.");
             }
 
+            var endPoint = serviceProvider.GetService<EndPoint>();
```

Comment:
I'm not a fan of having the EndPoint be directly in the container, it needs to hang off some option that gets passed to the HubConnection instead.

Reply:
I figured you might say that. Do you want to add a new HubConnectionOptions type with and EndPoint property?

I was thinking making a HubConnection.StartAsync take an EndPoint might be even better if we're willing to make that hard of a break

Reply:
That's a pretty awful API though. It would be more flexible though.

Reply:
Actually what you have isn’t so bad it gets bad only when you add this to a bigger service collection.

Reply:
I could see value in an `IOptions<HubConnectionOptions>` given the earlier discussions re constructor breaking changes... might be a safer approach.